.babelrc

```diff
@@ -1,20 +1,45 @@
 {
-    "presets": [
-        ["es2015", {
-            "modules": false
-        }],
-        "flow"
-    ],
-    "plugins": [
-        "transform-runtime",
-        "transform-class-properties",
-        "version-inline"
-    ],
+    "presets": ["flow"],
     "env": {
+        // Duplications are inevitable due to strange .babelrc env merging:
+        // babel/babel#5276. This can be simplified in Babel 7 using .babelrc.js.
+        "production": {
+            "presets": [["es2015", {"modules": false}]],
+            "plugins": [
+                "transform-class-properties",
+                "transform-runtime",
+                "version-inline",
+            ],
+        },
+        "development": {
+            "presets": [["es2015", {"modules": false}]],
+            "plugins": [
+                "transform-class-properties",
+                "transform-runtime",
+                "version-inline",
+            ],
+        },
         "test": {
+            "presets": ["es2015"],
+            "plugins": [
+                "transform-class-properties",
+                "transform-runtime",
+                "version-inline",
+            ],
+        },
+        "esm": {
             "plugins": [
-                "transform-es2015-modules-commonjs"
+                "transform-class-properties",
+                "version-inline",
+                ["import-rename", {
+                    // source files and font metrics data are moved to lib/
+                    "^\\./src/": "./lib/",
+                    "^\\.\\./submodules/katex-fonts/": "./",
+                    // file extensions are changed to .mjs
+                    // browsers are not able to resolve paths like Node
```

Comment:
It's too bad that `webpack` doesn't support outputting ES6 bundles.  I can't see very many people using `./lib/katex.js` in the browser since that will result in a bunch of additional requests.

Reply:
@kevinbarabash I'm not sure which use cases will be more common, using individual modules or using the main module.

If latter is more common, we can distribute bundled module using rollup.